### PR TITLE
[spec/interpreter/test] Land bulk instructions & reference types proposals

### PR DESCRIPTION
**Pull request by rossberg**
As per WG vote on 2021-02-10, both proposals have been promoted to phase 5 and are ready to be merged.

This PR includes both of them together, since they are mutually dependent. Technically, this is a merge of the reference-types repo, which already contained all parts of the bulk instructions repo, plus conflict resolutions and additions to fill in gaps.

### CODE REVIEW

**Comment by Ms2ger** [comment]
(The WG can't make sync decisions according to the charter, so I assume this is pending confirmation on the mailing list.)

**Comment by dschuff** [comment]
Thanks for bringing this up @Ms2ger 
The WG announcement has been made and the comment period has passed, so this should be good to go.

File: test/core/select.wast [hunk 1]
```diff
@@ -251,7 +315,7 @@
 (assert_return (invoke "as-br_table-last" (i32.const 1)) (i32.const 2))
 
 (assert_return (invoke "as-call_indirect-first" (i32.const 0)) (i32.const 3))
-(assert_return (invoke "as-call_indirect-first" (i32.const 1)) (i32.const 2))
+;;(assert_return (invoke "as-call_indirect-first" (i32.const 1)) (i32.const 2))
```
Comment (jblebrun) [inline test/core/select.wast:318]: Was it intentional that this was commented out?

Comment (rossberg) [reply test/core/select.wast:318]: Hm, probably not, but I can't trace back where it came from. Mind creating a PR to reactivate it?

Comment (jblebrun) [reply test/core/select.wast:318]: No problem, I've done so here: https://github.com/WebAssembly/spec/pull/1323

**Pull request merged**